File: 프로그래머스/lv0/120835. 진료 순서 정하기/진료 순서 정하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> emergency)
{
    vector<int> answer;
    answer = emergency;

    sort(emergency.rbegin(), emergency.rend());

    for (int i = 0; i < answer.size(); i++)
    {
        emergency[i] += 10;
        answer[i] += 10;
    }    

    for (int i = 0; i < emergency.size(); i++)
    {
        for (int j = 0; j < answer.size(); j++)
        {
            if (emergency[i] == answer[j])
                answer[j] = i + 1;
        }
    }

    return answer;
}

File: 프로그래머스/0/120835. 진료 순서 정하기/진료 순서 정하기.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] emergency) {
        int[] answer = new int[emergency.length];

        for (int i = 0, idx = 0, cnt = 0; i < emergency.length; i++) {
            int max = -1;
            for (int j = 0; j < emergency.length; j++) {
                if (emergency[j] > max) {
                    max = emergency[j];
                    idx = j;
                }
            }
            answer[idx] = ++cnt;
            emergency[idx] = -1;
        }

        return answer;
    }
}